src/nvim/buffer_defs.h

```diff
@@ -1158,6 +1158,12 @@ struct window_S {
    * In a non-location list window, w_llist_ref is NULL.
    */
   qf_info_T   *w_llist_ref;
+
+  /*
+   * Are we following the terminal? (Only valid if w_buffer is a terminal
+   * buffer.)
+   */
+  bool following_terminal;
```

Comment:
I haven't investigated this as fully as it deserves yet, but would it make sense to write a function, `window_should_follow()` determined by its current state, or do we have to determine this at a specific point?